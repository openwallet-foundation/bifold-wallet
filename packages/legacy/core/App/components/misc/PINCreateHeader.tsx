import { useTranslation } from 'react-i18next'
import { useTheme } from '../../contexts/theme'
import { View } from 'react-native'
import { ThemedText } from '../texts/ThemedText'

export interface PINCreateHeaderProps {
  updatePin?: boolean
}

const PINCreateHeader = ({ updatePin }: PINCreateHeaderProps) => {
  const { TextTheme } = useTheme()
  const { t } = useTranslation()
  return (
    <View>
      <ThemedText style={{ marginBottom: 16 }}>
        <ThemedText style={{ fontWeight: TextTheme.bold.fontWeight }}>
          {updatePin ? t('PINCreate.RememberChangePIN') : t('PINCreate.RememberPIN')}
        </ThemedText>{' '}
        {t('PINCreate.PINDisclaimer')}
      </ThemedText>
    </View>
  )
}

export default PINCreateHeader
