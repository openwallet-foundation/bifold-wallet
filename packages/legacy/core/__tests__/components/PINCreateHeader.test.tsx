import { render } from '@testing-library/react-native'
import React from 'react'

import PINCreateHeader from '../../App/components/misc/PINCreateHeader'

describe('PINCreateHeader Component', () => {
  test('Renders correctly', async () => {
    const tree = render(<PINCreateHeader />)

    expect(tree).toMatchSnapshot()
  })
})
