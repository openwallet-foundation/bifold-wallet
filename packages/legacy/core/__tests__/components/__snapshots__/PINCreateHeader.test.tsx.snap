// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PINCreateHeader Component Renders correctly 1`] = `
<View>
  <Text
    maxFontSizeMultiplier={2}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 18,
          "fontWeight": "normal",
        },
        Object {
          "marginBottom": 16,
        },
      ]
    }
  >
    <Text
      maxFontSizeMultiplier={2}
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "bold",
          },
        ]
      }
    >
      PINCreate.RememberPIN
    </Text>
     
    PINCreate.PINDisclaimer
  </Text>
</View>
`;
